### ISSUE TITLE
Document or maybe change default config for collation and charset in global.ini.php

### ISSUE THREAD

**Issue by mikkeschiren** [piwik/piwik#22721]
As default, Matomo is installed with charset [`utf8`](https://github.com/matomo-org/matomo/blob/5.x-dev/config/global.ini.php#L46) (which usually means `utf8mb3`), but the recommended charset is `utf8mb4`. 

For collation, the comment in default [config says](https://github.com/matomo-org/matomo/blob/5.x-dev/config/global.ini.php#L54) "Matomo should work correctly without this setting but we recommend to have a collation set" - but as of 5.1.2, this is needed to be set.

Should the default charset be changed to `utf8mb4`, and collation be set to `utf8mb4_unicode_ci`? Or maybe just update the comments in `global.ini.php` ?

**Comment by mikkeschiren** [comment]
I do realize that many still run Matomo on a non recommended charset - so it is challenge per se..

**Comment by sgiehl** [comment]
@mikkeschiren The default in config is `utf8`, as that should be supported by all supported database versions. During the installation we are trying to look up available charsets and if `utf8m4` is available, this value will be set to the local config and used instead of `utf8`

**Comment by mikkeschiren** [comment]
Yes, but should we update the comments in the global ini file? Because it is not entirly correct now.

**Comment by mikkeschiren** [comment]
Are there really any of the supported databases that doesn't support `utf8m4`?

**Comment by sgiehl** [comment]
officially we still support MySQL 5.5, which might not have supported utf8mb4 in the beginning.
But we might drop support for older MySQL releases in the next major version, so we might be able to update the config value then.
